actionpack/lib/action_dispatch/middleware/cookies.rb

```diff
@@ -569,7 +580,7 @@ def parse(name, signed_message)
         end
 
         def commit(options)
-          options[:value] = @verifier.generate(serialize(options[:value]))
+          options[:value] = @verifier.generate(serialize(options[:value]), expires_at: options[:expires_at], expires_in: options[:expires_in])
 
```

Comment:
@kaspth 

Currently, just the value gets serialized not the wrapped message with metadata since it uses NullSerailizer